Applications/SlicerApp/Testing/Python/CurvedPlanarReformation.py

```diff
@@ -0,0 +1,414 @@
+    should be such that other python code can import
+    this class and make use of the functionality without
+    requiring an instance of the Widget.
+    Uses ScriptedLoadableModuleLogic base class, available at:
+    https://github.com/Slicer/Slicer/blob/main/Base/Python/slicer/ScriptedLoadableModule.py
+    """
+
+    def __init__(self) -> None:
+        """Called when the logic class is instantiated. Can be used for initializing member variables."""
+        ScriptedLoadableModuleLogic.__init__(self)
+
+    def getParameterNode(self):
+        return CurvedPlanarReformationParameterNode(super().getParameterNode())
+
+    def process(
+        self,
+        inputVolume: vtkMRMLScalarVolumeNode,
+        outputVolume: vtkMRMLScalarVolumeNode,
+        imageThreshold: float,
+        invert: bool = False,
+        showResult: bool = True,
+    ) -> None:
+        """
+        Run the processing algorithm.
+        Can be used without GUI widget.
+        :param inputVolume: volume to be thresholded
+        :param outputVolume: thresholding result
+        :param imageThreshold: values above/below this threshold will be set to 0
+        :param invert: if True then values above the threshold will be set to 0, otherwise values below are set to 0
+        :param showResult: show output volume in slice viewers
+        """
+
+        if not inputVolume or not outputVolume:
+            raise ValueError("Input or output volume is invalid")
+
+        import time
+
+        startTime = time.time()
+        logging.info("Processing started")
+
+        # Compute the thresholded output volume using the "Threshold Scalar Volume" CLI module
+        cliParams = {
+            "InputVolume": inputVolume.GetID(),
+            "OutputVolume": outputVolume.GetID(),
+            "ThresholdValue": imageThreshold,
+            "ThresholdType": "Above" if invert else "Below",
+        }
+        cliNode = slicer.cli.run(slicer.modules.thresholdscalarvolume, None, cliParams, wait_for_completion=True, update_display=showResult)
+        # We don't need the CLI module node anymore, remove it to not clutter the scene with it
+        slicer.mrmlScene.RemoveNode(cliNode)
+
+        stopTime = time.time()
+        logging.info(f"Processing completed in {stopTime - startTime:.2f} seconds")
+
+
+#
+# CurvedPlanarReformationTest
+#
+
+
+class CurvedPlanarReformationTest(ScriptedLoadableModuleTest):
+    """
+    This is the test case for your scripted module.
+    Uses ScriptedLoadableModuleTest base class, available at:
+    https://github.com/Slicer/Slicer/blob/main/Base/Python/slicer/ScriptedLoadableModule.py
+    """
+
+    def setUp(self):
+        """Do whatever is needed to reset the state - typically a scene clear will be enough."""
+        slicer.mrmlScene.Clear()
+
+    def runTest(self):
+        """Run as few or as many tests as needed here."""
+        self.setUp()
+        self.test_CurvedPlanarReformation1()
+
```

Comment:
A dedicated test should be added ...

```suggestion

    def test_VolumeResamplerRegistered(self):
        resamplerName = "ResampleScalarVectorDWIVolume"
        self.assertIsNotNone(slicer.app.applicationLogic())
        self.assertTrue(appLogic.IsVolumeResamplerRegistered(resamplerName))
        
```